src/components/LessonCard/LessonCard.tsx

```diff
@@ -108,63 +76,192 @@ const Paragraph = styled(P)`
   line-height: 19px;
 `;
 
+const Hr = styled.hr`
+  margin: var(--spacing-x-large) 0 var(--spacing-x-large) 0;
+`;
+
 const StyledButton = styled(Button)``;
 
-type Props = {
+const selectMascot = (mascotType: MascotType, isResourcesCard: boolean, progress: Progress) => {
+  const hideHead = isResourcesCard ? false : (progress <= 2);
+  const hideBody = isResourcesCard ? false : (progress <= 1);
+  const hideLegs = isResourcesCard ? false : (progress <= 0);
+
+  switch (mascotType) {
+    case 1:
+      return (
+        <svg data-progress={progress} xmlns="http://www.w3.org/2000/svg" id="Robot-1" viewBox="0 0 225.53 341.04">
```

Comment:
I'm not sure I would be comfortable with burying logic/conditional inside an svg. An svg should just be plain static. Domain logic should be in the code.

In terms of the LessonCard owning that logic though - it was intentional by design in fact. So passing in an arbitrary url would mean the usage site will have to own that business logic. Which can still be fine, as long as we create some helper to facilitate that, so that Lessons/Resources/etc pages will not have to deal with sourcing those assets again themselves. 

Perhaps we should create a non UI component like AssetManager that loads all the assets, and then places like the LessonCard etc could just import that component, and then reference what they require (according to domain logic they should own and encapsulate). For example, LessonCard will reference `AssetManager.Shapes.LessonBackground1` when it needs the one for lesson 1, but it can remain the place to manage that logic, not on the usage site. Hope that makes sense.

Reply:
Similarly, though separately, also on the logic around showing additional graphics. I'm not sure it's a good idea to have those optional as part of the graphics themselves. Again those are business logic that should live on the higher up layer, the UI rendering side. 

So with my original vision of the LessonCard, this is actually going to be just another parameter/config that can be enabled to optionally show the ProgressIndicator component, below the mascot figure, using Flex layout to manage that layout logic.

Reply:
> I'm not sure I would be comfortable with burying logic/conditional inside an svg. An svg should just be plain static. Business logic should be in the code.
> 
> In terms of the LessonCard owning that logic though - it was intentional by design in fact. So passing in an arbitrary url would mean the usage site will have to own that business logic (and it may also be nothing to do with lessons!). Which can still be fine, as long as we create some helper to facilitate that, so that Lessons/Resources/etc pages will not have to deal with sourcing those assets again themselves.
> 
> Perhaps we should create a non UI component like AssetManager that loads all the assets, and then places like the LessonCard etc could just import that component, and then reference what they require (according to domain logic they should own and encapsulate). For example, LessonCard will reference `AssetManager.Shapes.LessonBackground1` when it needs the one for lesson 1, but it can remain the place to manage that logic, not on the usage site. Hope that makes sense.

On the logic/conditional inside svgs though, I'm really not sure this is the right way to go. Thoughts @owen26 ?

Reply:
my opinion is the same as my original comment, this SVG should be imported as an asset URL instead of a React component/node. This also aligns with my view on all images so far in this project. So that those assets are made compatible with external data persistence services such as a CMS. And they can be in other form not only SVG but JPEG and PNG.

Reply:
OK cool thanks. And just to be clear, my concerns were not just on the inline SVG, but also the use of the variables inside the SVG.

Reply:
Yes I agree. Inline SVG with variables is rarely used in practice. To be fair it can be powerful in specific use cases. But here however as I mentioned those SVG better be served as external assets rather than inline into component, so no variables involved.